src/main/java/com/google/api/generator/engine/ast/AssignmentExpr.java

```diff
@@ -59,6 +59,13 @@ public AssignmentExpr build() {
               String.format(
                   "LHS type %s must match RHS type %s", lhsType.toString(), rhsType.toString()));
         }
+      } else if (rhsType.isPrimitiveType()) {
+        if (!lhsType.equals(rhsType)) {
+          throw new TypeMismatchException(
+              String.format(
+                  "If LHS type %s is reference type and RHS type %s is primitive type, they should be boxed/primitive type equals.",
```

Comment:
Can we make this message shorter, e.g. "LHS and RHS primitive types are not equal"? The reader can infer the boxed equality implications.

Reply:
Sure, since in this case LHS type is reference type, and RHS is primitive type, updated it to make it clearer.
`LHS reference type %s and RHS primitive type %s are not equal.`